app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,263 @@
+  IDirectiveUsage,
+} from 'components/DirectiveInput/types';
+import { formatDirectiveUsageData } from 'components/DirectiveInput/utils';
+import T from 'i18n-react';
+import React, { useEffect, useRef, useState, useReducer } from 'react';
+import styled from 'styled-components';
+import {
+  PREFIX,
+  TWO_COLUMN_DIRECTIVE,
+  MULTIPLE_COLUMN_DIRECTIVE,
+} from 'components/DirectiveInput/constants';
+import { grey } from '@material-ui/core/colors';
+import CloseOutlinedIcon from '@material-ui/icons/CloseOutlined';
+import { reducer, initialDirectiveInputState } from 'components/DirectiveInput/reducer';
+
+const InputParentWrapper = styled(Box)`
+  display: block;
+  box-shadow: -3px -4px 15px rgba(68, 132, 245, 0.25);
+`;
+
+const SearchBarWrapper = styled(Box)`
+  padding: 10px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const InputWrapper = styled(Box)`
+  width: 100%;
+  display: flex;
+  align-items: center;
+`;
+
+const DirectiveUsageWrapper = styled(Box)`
+  background: ${grey[700]};
+`;
+
+const InputComponent = styled.input`
+  width: 95%;
+  margin-left: 5px;
+  outline: 0;
+  border: 0;
+  background: transparent;
+  color: #ffffff;
+`;
+
+const LabelComponent = styled.label`
+  color: #94ec98;
+  font-size: 14px;
+  margin-bottom: 0;
+`;
+
+const DirectiveBox = styled(Box)`
+  background-color: #ffffff;
+  position: fixed;
+  bottom: 93px;
+  width: 100%;
+`;
+
+const CloseIconButton = styled(IconButton)`
+  color: #ffffff;
+`;
+
+export default function({
+  columnNamesList,
+  onDirectiveInputHandler,
+  onClose,
+  openDirectivePanel,
+}: IDirectiveInputProps) {
+  const [directiveInput, dispatch] = useReducer(reducer, initialDirectiveInputState);
+  enum IDirectiveActions {
+    INPUT_DIRECTIVE,
+    DIRECTIVE_SET,
+    APPLIED_DIRECTIVE,
+    DIRECTIVE_COLUMN_COUNT,
+    DIRECTIVE_USAGE_LIST,
+    DIRECTIVE_LIST,
+    DIRECTIVE_PASTE,
+  }
+  const {
+    inputDirective,
+    isDirectiveSet,
+    directivesList,
+    appliedDirective,
+    directiveColumnCount,
+    isDirectivePaste,
+    directiveUsageList,
+  } = directiveInput;
+  const [directiveColumns, setDirectiveColumns] = useState([]);
+  const [enterCount, setEnterCount] = useState(0);
+  const directiveRef = useRef();
+
+  const handleDirectiveChange = (value) => {
+    if (!value) {
+      setEnterCount(0);
+      dispatch({
+        type: IDirectiveActions.INPUT_DIRECTIVE,
+        payload: false,
+      });
+    }
+    const inputText = value.split(' ');
+    if (directivesList.filter((el) => el.directive == inputText[0]).length) {
+      if (TWO_COLUMN_DIRECTIVE.includes(inputText[0])) {
+        dispatch({
+          type: IDirectiveActions.DIRECTIVE_COLUMN_COUNT,
+          payload: 2,
+        });
+      } else if (MULTIPLE_COLUMN_DIRECTIVE.includes(inputText[0])) {
+        dispatch({
+          type: IDirectiveActions.DIRECTIVE_COLUMN_COUNT,
+          payload: 0,
+        });
+      }
+      dispatch({
```

Comment:
Please do not dispatch multiple times like this - all of these should be contained in one dispatch